api/src/main/java/org/eclipse/microprofile/rest/client/annotation/RegisterClientHeaders.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.microprofile.rest.client.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.eclipse.microprofile.rest.client.ext.ClientHeadersFactory;
+import org.eclipse.microprofile.rest.client.ext.DefaultClientHeadersFactoryImpl;
+
+/**
+ * Used to specify that a {@link ClientHeadersFactory} should be used to generate or propagate HTTP headers on the outbound request.
+ * When annotation is placed at the interface level of a Rest Client interface, the implementation will invoke the ClientHeadersFactory's
+ * <code>update</code> method.
+ * <p>
+ * If no implementation class of the ClientHeadersFactory interface is specified in the annotation, then the
+ * {@link DefaultClientHeadersFactoryImpl} will be used.  This implementation will simply propagate headers (specified via MP Config property)
+ * from an inbound JAX-RS request (if applicable) to the outbound request.
+ * <p>
+ * If a ClientHeadersFactory listed is not found on the classpath, it is ignored.
```

Comment:
That line is intended to describe an error path behavior.  For example, if a user has a rest client interface like:
```
@RegisterClientHeaders(MyMissingHeaderFactory.class)
public interface MyServiceWithMissingHeaderFactory { ...
```

and `MyMissingHeaderFactory` is not packaged with the app - and the implementation cannot find/load it - then the behavior would be to continue the request without invoking the default client headers factory.  Ideally the implementation would also log a warning.

We could change this behavior so that if the class is missing, then we would throw a deployment exception - preventing the rest client instance from being created?  I am fine either way.

Reply:
I haven't looked at implementing it yet but I think from the user perspective it would be better if it failed with an exception. It should speed up spotting the problem.

Reply:
Sounds good.  I'll update and submit a new commit with this change.